proposals/0282-record-dot-syntax.rst

```diff
@@ -68,15 +69,39 @@ For the specification we focus on the changes to the parsing rules, and
 the desugaring, with the belief the type checking and renamer changes
 required are an unambiguous consequences of those.
 
-2.1 ``RecordDotSyntax`` language extension
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+2.1 Language extensions
+~~~~~~~~~~~~~~~~~~~~~~~
 
-This change adds a new language extension ``RecordDotSyntax``.
+This change adds new language extensions ``OverloadedRecordDot`` and
+``OverloadedRecordUpdate``.
+
+The effect of ``OverloadedRecordDot`` is to:
+  - allow ``.`` in record expressions;
+
+    - this includes expressions ``r.x`` and updates ``r{a.b = e}``;
+  - desguars field access via ``.`` to ``getField`` expressions;
+The effects of ``OverloadedRecordUpdate`` are to:
+  - desguar record updates to ``setField`` expressions;
+  - disallow the use of qualified labels in updates ``r{M.x = 1}`` is a syntax error
+
+This means:
+  - When ``OverloadedRecordDot`` is not enabled:
+
+    - ``e{lbl₁.lbl₂ = val}`` is a syntax error.
+  - When ``OverloadedRecordUpdate`` is enabled without ``OverloadedRecordDot``:
+
+    - ``e{lbl₁} = val`` will desguar to a ``setField`` expression;
+    - ``e{lbl₁.lbl₂ = val}`` is a syntax error.
+  - When both ``OverloadedRecordDot`` and ``OverloadedRecordUpdate`` are enabled:
```

Comment:
When ``OverloadedRecordDot`` is enabled without ``OverloadedRecordUpdate``:
    - ``e{lbl₁} = val`` is treated exactly as in Haskell 98
    - ``e{lbl₁.lbl₂ = val}`` is rejected